src/ivcam/sr300.cpp

```diff
@@ -339,24 +339,33 @@ namespace librealsense
         using namespace std;
         using namespace std::chrono;
 
-        try {
-            command cmd(ivcam::GoToDFU);
+        try
+        {
+            command cmd( ivcam::GoToDFU );
             cmd.param1 = 1;
-            _hw_monitor->send(cmd);
-            std::vector<uint8_t> gvd_buff(HW_MONITOR_BUFFER_SIZE);
-            for (auto i = 0; i < 50; i++)
+            _hw_monitor->send( cmd );
+            std::vector< uint8_t > gvd_buff( HW_MONITOR_BUFFER_SIZE );
+            // 120 iterations = ~6 seconds, we allow 6 seconds because on Linux the removal status is
+            // updated at a 5 seconds rate.
+            for( auto i = 0; i < 120; i++ )
             {
-                _hw_monitor->get_gvd(gvd_buff.size(), gvd_buff.data(), ds::GVD);
-                this_thread::sleep_for(milliseconds(50));
+                // If the device was detected as removed we assume the device is entering update
+                // mode
+                if( ! is_valid() )
+                    return;
+
+                this_thread::sleep_for( milliseconds( 50 ) );
             }
-            throw std::runtime_error("Device still connected!");
+
+            LOG_WARNING( "Device still connected!" );
```

Comment:
I think we might want some better explanation in the logs...
Waited 6 seconds after `enter_update_state` but device is still connected!

Reply:
Changed to "Timeout waiting for device disconnect after DFU command!" OK?